### ISSUE TITLE
Null State when grain interface and implementation are in the same project

### ISSUE THREAD

**Issue by kellermanrivero** [dotnet/orleans#562]
How to reproduce this issue:

* Create Class Library Project.
* Install the Microsoft.Orleans.Templates.Grains and Microsoft.Orleans.Templates.Templates nuget packages in the class library project.
* Create an grain interface and implementation. 
* Create a Test Local Silo
* Invoke method on the grain, State property is null.

**Comment by jthelin** [comment]
Your grain class and interface projects should only use one of the Template packages, not both.

The normal workflow is:

-- Create a C# library project for grain interface(s). 
-- Install the Microsoft.Orleans.Template.GrainInterafaces nuget package in this first project.
-- Write a grain interface declaration [C# interface].

-- Create a C# library project for grain class(es).
-- Install the Microsoft.Orleans.Template.Grains nuget package in that second project.
-- Write a grain implementation class which extends Grain<T> [C# class]. 

I know it was talked about sometime in the past, but I don't recall if combined interface+grain projects are a supported scenario.

**Comment by gabikliot** [comment]
It used to be not supported, by design, when we used concrete code-gened GetGrain calls on grain factories. One could not define grain interface for IMyGrain and use MyGrainFactory.GetGrain in the same project.
Now as we deprecated concrete code-gened GetGrain calls (#540) in favor of generic factories, defining both interfaces and impl. in the same project should in theory be possible, but there might be some other issue blocking it.
